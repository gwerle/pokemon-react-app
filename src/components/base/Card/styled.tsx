import styled, { css } from "styled-components";

export const CardContainer = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xsmall};
    border: ${`1px solid ${theme.colors.darkRed}`};
    font-size: ${theme.font.sizes.xsmall};
    border-radius: 10px;
    box-shadow: 0 0 0.2rem ${theme.colors.lightGray};
  `}
`;
