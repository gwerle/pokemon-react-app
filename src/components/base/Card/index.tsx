import Skeleton from "../Skeleton";
import * as S from "./styled";

type Props = {
  children: JSX.Element;
  imgSrc: string;
  isLoading: boolean;
};

export default function Card({ children, imgSrc, isLoading }: Props) {
  return (
    <S.CardContainer>
      {isLoading ? (
        <Skeleton height="150px" width="100%" />
      ) : (
        <img src={imgSrc} alt={"pokemon"} height="150px" />
      )}
      {children}
    </S.CardContainer>
  );
}
