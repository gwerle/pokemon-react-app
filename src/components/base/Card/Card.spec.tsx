import { screen } from "@testing-library/react";

import { renderWithTheme } from "@/utils/tests/renderWithTheme";

import Card from ".";

describe("<Card />", () => {
  it("Renders without error", () => {
    const { container } = renderWithTheme(
      <Card imgSrc="https://google.com" isLoading={false}>
        <div data-testid="card-child">child</div>
      </Card>
    );

    expect(container).toMatchSnapshot();
  });

  it("should show loading", () => {
    renderWithTheme(
      <Card imgSrc="https://google.com" isLoading={true}>
        <div data-testid="card-child">child</div>
      </Card>
    );

    const loading = screen.getByTestId("skeleton");

    expect(loading).toBeInTheDocument();
  });

  it("should show child", () => {
    renderWithTheme(
      <Card imgSrc="https://google.com" isLoading={false}>
        <div data-testid="card-child">child</div>
      </Card>
    );

    const loading = screen.getByTestId("card-child");

    expect(loading).toBeInTheDocument();
  });
});
