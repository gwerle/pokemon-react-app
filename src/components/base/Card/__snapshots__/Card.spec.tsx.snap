// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Renders without error 1`] = `
<div>
  <div
    class="sc-gueYoa emUOiO"
  >
    <img
      alt="pokemon"
      height="150px"
      src="https://google.com"
    />
    <div
      data-testid="card-child"
    >
      child
    </div>
  </div>
</div>
`;
